xdsl/transforms/snrt_to_riscv_asm.py

```diff
@@ -63,10 +63,103 @@ def match_and_rewrite(
+                        (    0b000 << 12) | \
+                        (0b0101011 <<  0)   \n" ::"r"(reg_src_high),
+                    "r"(reg_src_low));
+
+                // dmdst a0, a1
+                asm volatile(
+                    ".word (0b0000001 << 25) | \
+                        (     (11) << 20) | \
+                        (     (10) << 15) | \
+                        (    0b000 << 12) | \
+                        (0b0101011 <<  0)   \n" ::"r"(reg_dst_high),
+                    "r"(reg_dst_low));
+
+                // dmcpyi a0, a4, 0b00
+                register uint32_t reg_txid asm("a0");  // 10
+                asm volatile(
+                    ".word (0b0000010 << 25) | \
+                        (  0b00000 << 20) | \
+                        (     (14) << 15) | \
+                        (    0b000 << 12) | \
+                        (     (10) <<  7) | \
+                        (0b0101011 <<  0)   \n"
+                    : "=r"(reg_txid)
+                    : "r"(reg_size));
+
+                return reg_txid;
+            } else {
+                return -1;
+            }
+        }
+
+        P.S. We only implement taking the top branch for now.
+        """
+        reg_t = riscv.IntRegisterType.unallocated()
+        rewriter.replace_matched_op(
+            [
+                # "Take an ui64 and split it in two 32 bit-wide RISC-V registers"
+                split_i64_dst := builtin.UnrealizedConversionCastOp.get(
+                    [op.dst],
+                    [reg_t, reg_t],
+                ),
+                # "Take an ui64 and split it in two 32 bit-wide RISC-V registers"
+                split_i64_src := builtin.UnrealizedConversionCastOp.get(
+                    [op.src],
+                    [reg_t, reg_t],
+                ),
+                # "Convert an IR-level i32 to a RISC-V register"
+                i32_size := builtin.UnrealizedConversionCastOp.get(
+                    [op.size],
+                    [reg_t],
+                ),
+                riscv_snitch.DMSourceOp(
+                    split_i64_src.results[0], split_i64_src.results[1]
+                ),
+                riscv_snitch.DMDestinationOp(
+                    split_i64_dst.results[0], split_i64_dst.results[1]
+                ),
+                copy_imm := riscv_snitch.DMCopyImmOp(i32_size, 0),
+                tx_id := builtin.UnrealizedConversionCastOp.get(
+                    [copy_imm], [builtin.i32]
+                ),
+            ],
+            new_results=tx_id.results,
+        )
+
+
 class ConvertSnrtToRISCV(ModulePass):
     name = "convert-snrt-to-riscv-asm"
 
     def apply(self, ctx: MLContext, op: builtin.ModuleOp) -> None:
         PatternRewriteWalker(
-            GreedyRewritePatternApplier([LowerClusterHWBarrier(), LowerSSRDisable()])
+            GreedyRewritePatternApplier(
+                [LowerClusterHWBarrier(), LowerSSRDisable(), LowerDMAStart1DWidePtr()]
```

Comment:
Just to make the diff a little neater in the future, could you please add a trailing comma here? This will split the patterns to individual lines
```suggestion
                [LowerClusterHWBarrier(), LowerSSRDisable(), LowerDMAStart1DWidePtr(),]
```